src/gui/guiBox.cpp

```diff
@@ -31,8 +34,82 @@ void GUIBox::draw()
 	if (!IsVisible)
 		return;
 
-	Environment->getVideoDriver()->draw2DRectangle(m_color, AbsoluteRect,
-			&AbsoluteClippingRect);
+	std::vector<s32> inner_bd = {0, 0, 0, 0};
+	std::vector<s32> outer_bd = {0, 0, 0, 0};
+
+	for (unsigned int i = 0; i <= 3; i++) {
+		if (m_borderwidths[i] > 0)
+			outer_bd[i] = m_borderwidths[i];
+		else
+			inner_bd[i] = m_borderwidths[i];
+	}
+
+	v2s32 pos = AbsoluteRect.UpperLeftCorner;
+	v2s32 geom = AbsoluteRect.LowerRightCorner;
+
+	v2s32 bd_pos = {
+		pos.X - outer_bd[3],
+		pos.Y - outer_bd[0]
+	};
+
+	v2s32 rect_pos = {
+		pos.X - inner_bd[3],
+		pos.Y - inner_bd[0]
+	};
+
+	v2s32 bd_geom = {
+		geom.X + outer_bd[1],
+		geom.Y + outer_bd[2]
+	};
+	v2s32 rect_geom = {
+		geom.X + inner_bd[1],
+		geom.Y + inner_bd[2]
+	};
+
+	core::rect<s32> main_rect(
+		rect_pos.X,
+		rect_pos.Y,
+		rect_geom.X,
+		rect_geom.Y
+	);
+
+	std::vector<core::rect<s32>> border_rects(4);
```

Comment:
Linked list abuse.
`core::rect<s32> border_rects[4];` is good enough.

Reply:
C++ ruins my brain.  I don't know why I don't do these things.

Reply:
Vectors don't use linked list, and a std::array is usually preferable to a raw array